src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3258,10 +3257,36 @@ private static bool TryGetNumberValue(string strNum, bool hex, bool real, char s
                         return false;
                     }
 
+                    if (suffix == 'u' || suffix == 'U')
```

Comment:
But... My inclination for that would almost be that it would be better for it to either throw or produce an unsigned doubl, if there is such a thing.

Reply:
Also, upon reviewing the code paths for `l` suffix... there are *two* of them. It seems a bit like there's a double-up on code paths there.

I'm not sure how the `real` boolean is determined, but `l` suffix has a code path inside the `double` parsing segment for that as well as one outside and after it.

Reply:
Been thinking on this, and I think that it would *probably* be best to maintain parity with the existing `L` token suffix. 

Adding a small branch for that code path in the `if (real){}` block as well, mimicking the doubling up the `long` parsing has. (I do wish there was a good way to consolidate them, but it looks like we're either gonna be doubling `real` checks for every parse or just doubling a couple of parse checks).

Reply:
And done!

Had a slip up where I forgot my `u` token checks were before the check for `real` which caused a slight issue for a moment.

Also added a test or two to check the round-on-parse behaviour.